### PR TITLE
Adding  branch-name copy  to clipboard branches screen.

### LINKED ISSUE #25120: Add "copy branch name" (icon) on dropdown branch selector
### Feature Description

I want to have ability to easy copy branch name with dropdown menu. Now i can only do this from URL link after selecting the necessary branch

I expect to add this copy button as on the attached mock.
Also, add a context menu "copy branch name" when you click on ... button (button not yet available/implemented)

### Screenshots

image

### PR DESCRIPTION
**Pull request by puni9869**
Adding  branch-name copy  to clipboard and button in branches screen

Replaces #25569
Fixes #25120

New mocks:
Screenshot 2023-06-30 at 12 01 41 AM
Screenshot 2023-06-30 at 12 03 59 AM

image

### CODE REVIEW

**Comment by puni9869** [comment]
@silverwind @lunny @wxiaoguang

File: templates/repo/branch/list.tmpl [hunk 1]
```diff
@@ -22,7 +22,10 @@
 								{{if .DefaultBranchBranch.IsProtected}}
 									{{svg "octicon-shield-lock"}}
 								{{end}}
-								<a href="{{.RepoLink}}/src/branch/{{PathEscapeSegments .DefaultBranchBranch.DBBranch.Name}}">{{.DefaultBranchBranch.DBBranch.Name}}</a>
+								<div class="branch-detail">
```
Comment (wxiaoguang) [inline templates/repo/branch/list.tmpl]: `class="flex-text-block"`

File: templates/repo/branch/list.tmpl [hunk 2]
```diff
@@ -22,7 +22,10 @@
...
+									<a class="branch-link" href="{{.RepoLink}}/src/branch/{{PathEscapeSegments .DefaultBranchBranch.DBBranch.Name}}">{{.DefaultBranchBranch.DBBranch.Name}}</a>
```
Comment (wxiaoguang) [inline templates/repo/branch/list.tmpl]: `class="gt-ellipsis"`

**Review by wxiaoguang — COMMENTED** [review]
There could be some existing CSS classes for the styles.

File: templates/repo/branch/list.tmpl (hunk 1 above)
Comment (puni9869) [reply templates/repo/branch/list.tmpl]: Problem:- 
image

`flex-wrap` is creating problem

image

Comment (wxiaoguang) [reply templates/repo/branch/list.tmpl]: Then use `class="gt-df gt-ac"` instead ? These are single style helper CSS classes.

Comment (puni9869) [reply templates/repo/branch/list.tmpl]: From where I can get this cheat sheet.

Comment (wxiaoguang) [reply templates/repo/branch/list.tmpl]: some are in `helper.css`

some are on the "devtest" page ( `http://localhost:3000/devtest`  in dev mode)

**Comment by puni9869** [comment]
Now I guess the classes are fine.

File: templates/repo/branch/list.tmpl [hunk 3]
```diff
@@ -76,14 +79,20 @@
 							<tr>
 								<td class="eight wide">
 								{{if .DBBranch.IsDeleted}}
-									<s><a href="{{$.RepoLink}}/src/branch/{{PathEscapeSegments .DBBranch.Name}}">{{.DBBranch.Name}}</a></s>
+									<div class="gt-df gt-ac">
+										<a class="gt-ellipsis" href="{{$.RepoLink}}/src/branch/{{PathEscapeSegments .DBBranch.Name}}">{{.DBBranch.Name}}</a>
+										<button class="btn interact-fg gt-p-3" data-clipboard-text="{{.DBBranch.Name}}">{{svg "octicon-copy" 14}}</button>
+									</div>
```
Comment (silverwind) [inline templates/repo/branch/list.tmpl:85]: You can probably avoid the extra wrapper via `gt-ac gt-ellipsis` on the `<a>`.

**Comment by silverwind** [comment]
Wow, this table looks so bad in those screenshots, but probably not this PRs fault (alone). Please provide comparison before/after screenshots here to ensure this didn't regress it further.

File: templates/repo/branch/list.tmpl (hunk 3 above)
Comment (puni9869) [reply templates/repo/branch/list.tmpl:85]: Not possible
image

image

we want the whole flex-box div item to be in centered alinged.
And if we apply `gt-ac`

image

The branch anchor tag is perfectly centered.

**Comment by puni9869** [comment]
> Wow, this table looks so bad in those screenshots, but probably not this PRs fault (alone). Please provide comparison before/after screenshots here to ensure this didn't regress it further.

💯 👍 😄 
Honestly, I am very happy that we do care about UI/UX a lot. But existing UI of gitea need a lot of improvements. IMO a dedicated milestone is required.

**Comment by puni9869** [comment]
> Wow, this table looks so bad in those screenshots, but probably not this PRs fault (alone). Please provide comparison before/after screenshots here to ensure this didn't regress it further.

My soul is also hearting when I see `<tr><td></td></tr>` , in the world of `flex` and `grid` and with specially in the mobile layout.

Here is the before and after comparison.

Before
image

Mobile view:
image

image


After  applying copy button:
image


Mobile view:
image

image


IMO we should change `<table>` layout in next followup PR. Let this PR merge if this address the required ISSUE fix.
Otherwise we will lose the track.

**Review by lunny — APPROVED** [review]
Approve the copy button position.

**Review by silverwind — APPROVED** [review]

**Pull request merged**